modules/javafx.graphics/src/main/java/com/sun/javafx/scene/NodeHelper.java

```diff
@@ -65,23 +63,7 @@ protected NodeHelper() {
     }
 
     protected static NodeHelper getHelper(Node node) {
-
-        NodeHelper helper = nodeAccessor.getHelper(node);
-        if (helper == null) {
-            String nodeType;
-            if (node instanceof Shape) {
-                nodeType = "Shape";
-            } else if (node instanceof Shape3D) {
-                nodeType = "Shape3D";
-            } else {
-                nodeType = "Node";
-            }
-
-            throw new UnsupportedOperationException(
-                    "Applications should not extend the "
-                    + nodeType + " class directly.");
-        }
-        return helper;
+        return nodeAccessor.getHelper(node);
```

Comment:
I think that would be unnecessarily verbose, and it doesn't seem like it would offer a benefit: as of now, this can't happen. Let's assume that the class hierarchy is changed in the future, such that `null` might be returned. This wouldn't result in a compile-time error. The exception would only be thrown when someone tries to acquire the `NodeHelper`, but in this case, we'd most likely get a NPE anyway.